### PR TITLE
[12.0][MIG] stock_scanner_inventory

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

**Review by gurneyalex — APPROVED** [review]

**Comment by eantones** [comment]
Hey @rousseldenis 
the `set_tmp_value` method is deprecated in `stock_scanner` v12
https://github.com/OCA/stock-logistics-barcode/blob/633a49cc6f638d8428c83307b0bd531b9aa14607/stock_scanner/models/scanner_hardware.py#L248-L256

I think we can take advantatge of this PR and get it upgraded.

Thanks!

File: stock_scanner_inventory/data/Inventory/scanner_scenario_step_inventory_product_code.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+from odoo import fields
+
+# No current inventory, create a new one
+if not terminal.get_tmp_value('tmp_val1'):
+    stock_inventory_obj = env['stock.inventory']
+    stock_inventory_id = stock_inventory_obj.create({
+        'name': '%s : %s' % (terminal.code, fields.Datetime.now()),
+    })
+    terminal.set_tmp_value('tmp_val1', stock_inventory_id.id)
```
Comment (eantones) [inline stock_scanner_inventory/data/Inventory/scanner_scenario_step_inventory_product_code.py:16]:
```suggestion
    terminal.update_tmp_values({'tmp_val1': stock_inventory_id.id})
```

File: stock_scanner_inventory/data/Inventory/scanner_scenario_step_inventory_product_code.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    stock_inventory_line_obj.create({
+        'inventory_id': stock_inventory_id,
+        'product_id': product.id,
+        'product_uom_id': product.uom_id.id,
+        'product_qty': quantity,
+        'location_id': location.id,
+    })
+    terminal.set_tmp_value('tmp_val2','')
+    terminal.set_tmp_value('tmp_val3','')
```
Comment (eantones) [inline stock_scanner_inventory/data/Inventory/scanner_scenario_step_inventory_product_code.py:36]:
```suggestion
    terminal.clean_tmp_values(['tmp_val2', 'tmp_val3'])
```

File: stock_scanner_inventory/data/Inventory/scanner_scenario_step_inventory_product_quantity.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# flake8: noqa
+# Use <m> or <message> to retrieve the data transmitted by the scanner.
+# Use <t> or <terminal> to retrieve the running terminal browse record.
+# Put the returned action code in <act>, as a single character.
+# Put the returned result or message in <res>, as a list of strings.
+# Put the returned value in <val>, as an integer
+
+terminal.set_tmp_value('tmp_val2', message)
```
Comment (eantones) [inline stock_scanner_inventory/data/Inventory/scanner_scenario_step_inventory_product_quantity.py:8]:
```suggestion
terminal.update_tmp_values({'tmp_val2': message})
```

**Comment by zeroheure** [comment]
Simple changes are needed,k any news ?

**Pull request closed without merging**